Multisig/UI/ClaimToken/ClaimingAmountViewController/ClaimTokensViewController.swift

```diff
@@ -0,0 +1,496 @@
+            claimButtonContainer.trailingAnchor.constraint(equalTo: claimButton.trailingAnchor, constant: 16),
+            claimButton.topAnchor.constraint(equalTo: claimButtonContainer.topAnchor, constant: 8),
+            claimButtonContainer.bottomAnchor.constraint(equalTo: claimButton.bottomAnchor, constant: 20),
+            claimButton.heightAnchor.constraint(equalToConstant: 56)
+        ])
+
+        // need to remember in order to modify bottom spacing when keyboard is shown or hidden
+        claimButtonBottom = view.safeAreaLayoutGuide.bottomAnchor.constraint(equalTo: claimButtonContainer.bottomAnchor)
+
+        // inject the button container below tableView and above the bottom of the screen
+
+            // reset table view layout
+        let tableViewConstraints = view.constraints.filter { ($0.firstItem as? UITableView) == tableView }
+        NSLayoutConstraint.deactivate(tableViewConstraints)
+
+        view.addConstraints([
+            // re-attach table view
+            tableView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+            tableView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+            tableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            tableView.bottomAnchor.constraint(equalTo: claimButtonContainer.topAnchor),
+
+            // attach button container
+            claimButtonContainer.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+            claimButtonContainer.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+            claimButtonBottom
+        ])
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        tableView.flashScrollIndicators()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        keyboardBehavior.start()
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        keyboardBehavior.stop()
+    }
+
+    @objc func didBeginEditing(_ notification: NSNotification) {
+        guard let textField = notification.object as? UITextField else { return }
+        keyboardBehavior.activeTextField = textField
+    }
+
+    // claim & delegate
+    @objc func didTapClaimButton() {
+        guard
+            let _ = inputAmount,
+            let _ = claimData,
+            let _ = safe,
+            let _ = delegateAddress,
+            let _ = timestamp
+        else {
+            return
+        }
+
+        completion()
+    }
+
+    var isMax: Bool {
+        guard
+            let row = rows.firstIndex(of: .claimingAmount),
+            let cell = tableView.cellForRow(at: IndexPath(row: row, section: 0)) as? ClaimedAmountInputCell
+        else {
+            return false
+        }
+        return cell.isMax
+    }
+
+    var inputAmount: Sol.UInt128? {
```

Comment:
this must be a controller's state and not the state of a view which can be re-created